[Bug]: aws_opesearch_domain data source not working in 1.40.0 release

### Terraform Core Version

1.6.3

### AWS Provider Version

5.40.0

### Affected Resource(s)

data aws_opensearch_domain

### Expected Behavior

Data source will return a correct data block as per the documentation

### Actual Behavior

An error is returined.



### Relevant Error/Panic Output Snippet

```shell
Error: setting auto_tune_options: Invalid address to set: []string{"auto_tune_options", "0", "use_off_peak_window"}
```


### Terraform Configuration Files

data "aws_opensearch_domain" "local_domain" {
  provider = aws.master
  domain_name     = "${var.environment}-elk"
}

### Steps to Reproduce

Use AWS version 1.40.0 to get the bug.
Roll back to 5.39.1 and the bug goes away.

### Debug Output

  http.response.header.x_amzn_requestid=8accff94-4235-4664-a111-2293abce6909 rpc.method=DescribeDomainConfig rpc.system=aws-api tf_data_source_type=aws_opensearch_domain aws.region=us-west-2 http.response.header.content_type=application/json rpc.service=OpenSearch tf_req_id=3cded91b-e7c4-8fcc-dc0e-2e76134ef29e @module=aws http.response_content_length=5323 tf_mux_provider="*schema.GRPCProviderServer" tf_rpc=ReadDataSource http.duration=49 http.response.header.date="Mon, 11 Mar 2024 02:26:41 GMT" http.status_code=200 tf_aws.sdk=aws-sdk-go tf_provider_addr=registry.terraform.io/hashicorp/aws timestamp=2024-03-11T02:26:41.801Z
2024-03-11T02:26:41.802Z [ERROR] provider.terraform-provider-aws_v5.40.0_x5: [ERROR] setting state: Invalid address to set: []string{"auto_tune_options", "0", "use_off_peak_window"}
2024-03-11T02:26:41.802Z [ERROR] provider.terraform-provider-aws_v5.40.0_x5: Response contains error diagnostic: diagnostic_severity=ERROR @caller=github.com/hashicorp/terraform-plugin-go@v0.22.0/tfprotov5/internal/diag/diagnostics.go:58 diagnostic_detail="" diagnostic_summary="setting auto_tune_options: Invalid address to set: []string{\"auto_tune_options\", \"0\", \"use_off_peak_window\"}" tf_data_source_type=aws_opensearch_domain tf_req_id=3cded91b-e7c4-8fcc-dc0e-2e76134ef29e tf_rpc=ReadDataSource tf_proto_version=5.4 @module=sdk.proto tf_provider_addr=registry.terraform.io/hashicorp/aws timestamp=2024-03-11T02:26:41.801Z

### Panic Output

_No response_

### Important Factoids

_No response_

### References

_No response_

### Would you like to implement a fix?

No

--- a/.changelog/36298.txt
+++ b/.changelog/36298.txt
@@ -1,3 +1,3 @@
 ```release-note:enhancement
-data-source/aws_opensearch_domain : Add missing `use_off_peak_window` attribute that caused an "invalid address" error
+data-source/aws_opensearch_domain : Add `auto_tune_options.use_off_peak_window` attribute. This fixes a regression introduced in [v5.40.0](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md#5400-march--7-2024) causing `Invalid address to set` errors
 ```

--- a/.changelog/36298.txt
+++ b/.changelog/36298.txt
@@ -1,0 +1,3 @@
+```release-note:enhancement
+data-source/aws_opensearch_domain : Add missing `use_off_peak_window` attribute that caused an "invalid address" error
+```

--- a/internal/service/opensearch/domain_data_source.go
+++ b/internal/service/opensearch/domain_data_source.go
@@ -101,6 +101,10 @@ func DataSourceDomain() *schema.Resource {
 							Type:     schema.TypeString,
 							Computed: true,
 						},
+						"use_off_peak_window": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
 					},
 				},
 			},

